### PR TITLE
Add git hook to run esLinst before commit

### PR DESCRIPTION
**Pull request by rdogra-nimble**
This PR covers the changes which has been done to make esLint check run before  any commit.

run eslint before committing the changes

### CODE REVIEW

File: end-to-end-tests/package.json [hunk 1]
```diff
@@ -6,7 +6,10 @@
     "test": "echo \"Error: no test specified\" && exit 1",
     "cypress:open": "cypress open",
     "cypress:run": "cypress run",
-    "lint": "eslint . --ext .cy.js,.jsx,.ts,.tsx --ignore-path .gitignore"
+    "lint": "eslint . --ext .cy.js,.jsx,.ts,.tsx --ignore-path .gitignore",
+    "precommit": "npm run lint",
```
Comment (jordan-beard) [inline end-to-end-tests/package.json:10]: why do you need the precommit script?

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm run precommit
```
Comment (jordan-beard) [inline .husky/pre-commit:4]: Could you change this to `npm run lint` and remove `npm run precommit` from the scripts?

File: package.json [hunk 3]
```diff
@@ -4,6 +4,25 @@
...
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.cy.js": [
+      "npm run precommit"
+    ]
+  },  
+  "scripts": {
+    "lint-staged": "npm lint-staged",
```
Comment (jordan-beard) [inline package.json:23]: Is this right? it looks like its defining itself 🤔

File: .husky/pre-commit (hunk 2 above)
Comment (rdogra-nimble) [reply .husky/pre-commit:4]: precommit is the script defined in the package.json which is actuall `npm run lint` to mention what we expect to run before commit the changes . Also precommit is a file which is created in the .husky folder.
I'll try to make the changes you suggested and see if it still works :)

**Review by jordan-beard — APPROVED** [review]

**Pull request merged**